whirlpool/appliance.py

```diff
@@ -146,56 +103,53 @@ async def send_attributes(self, attributes):
                     LOGGER.error(f"Sending attributes failed ({r.status})")
         return False
 
-    def get_attribute(self, attribute):
+    def register_attr_callback(self, update_callback: Callable):
+        """Register Callback function."""
+        self._attr_changed.append(update_callback)
+        LOGGER.debug("Registered attr callback")
+
+    def unregister_attr_callback(self, update_callback: Callable):
+        """Unregister callback function."""
+        try:
+            self._attr_changed.remove(update_callback)
+            LOGGER.debug("Unregistered attr callback")
+        except ValueError:
+            LOGGER.error("Attr callback not found")
+
+    def update_attributes(self, attrs: dict[str, str], timestamp: str):
+        for attr, val in attrs.items():
+            if self.has_attribute(attr):
+                self._set_attribute(attr, val, timestamp)
```

Comment:
```suggestion
                self._set_attribute(attr, str(val), timestamp)
```
since attrs can be `Any`, this needs to cast to str again